hack/release.sh

```diff
@@ -14,18 +14,19 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-set -ex
+source $(dirname $0)/../vendor/github.com/knative/test-infra/scripts/release.sh
 
-pushd $(dirname "$0")/..
+function build_release() {
+  # Run `generate-yamls.sh`, which should be versioned with the
+  # branch since the detail of building may change over time.
+  local YAML_LIST="$(mktemp)"
```

Comment:
Is there a way we can not use mktemp as that will leak files and it is hard to ensure cleanups in bash. Maybe:

```bash
if [ -f "${REPO_ROOT_DIR}/.serving-operator-temp.yaml" ];
  echo "Error: ${REPO_ROOT_DIR}/.serving-operator-temp.yaml already exists!"
  exit 1
fi

YAML_LIST="${REPO_ROOT_DIR}/.serving-operator-temp.yaml"
```

and then mv this file over when done?

Reply:
I guess I can simplify it by using the HOME_DIR of our repo, and put serving-operator.yaml directly under it.

Reply:
If does now matter whether .serving-operator-temp.yaml exists or not.
It will be either created or overwritten, when release.sh is called.